doc/HowToPipeline.md

```diff
@@ -0,0 +1,2040 @@
+`connections` attribute is a sub-config that has fields corresponding to the variable names of the
+connections, with values of those fields corresponding to the name of a connection. So by default
+`config.connections.outputCatalog` would be `customAperture` and `config.connections.exposure` would be
+`calexp` etc. Assigning to these config fields has the effect of changing the name of the data-set type
+defined in the connection.
+
+In this config file we are changing the name of the data-set type that will be persisted to include the
+information that local background subtraction was done. It is interesting to note that there are no hard
+coded data-set type names that must be adhered to, the user is free to pick any name. The only consequence
+of changing a data-set type name, is that any down stream code that is to use the output data-set must have
+its default name changed to match. As an aside, the current activator requires that the first time a
+data-set name is used the activator command is run with the `--register-dataset-types` switch. This is to
+prevent accidental typos becoming new data-set types.
+
+Looking at the config file, there are two different fields that we are setting to the same value. In the
+case of other tasks this number may even be higher. This leads not only to the issue of config overrides
+needing to potentially be lengthy, but that there may be a typo, and the fields will be set inconsistently.
+To address this, `PipelineTasks` have a way to template data-set type names. Let's see what this looks
+like.
+
+```python
+class ApertureTaskConnections(pipeBase.PipelineTaskConnections,
+                              defaultTemplates={"outputName":"customAperture"},
+                              dimensions=("visit", "detector", "abstract_filter", "skymap")):
+    ...
+
+    outputSchema = pipeBase.connectionTypes.InitOutput(doc="Schema generated in Aperture PipelineTask",
+                                                       storageClass="SourceCatalog",
+                                                       name="{outputName}_schema")
+    
+    ...
+
+    outputCatalog = pipeBase.connectionTypes.Output(doc="Aperture measurements",
+                                                    dimensions=("visit", "detector", "abstract_filter",
+                                                                "skymap"),
+                                                    storageClass="SourceCatalog",
+                                                    name="{outputName}")
+```
+
+In the modified connection class, the `outputSchema` and `outputCatalog` connections now have python format
+strings, which are referred to with the template name `outputName`. This template will formatted a
+specified string to become a data-set type name prior to any data being read and supplied to the task. The
+class declaration also has a new argument along side the `dimensions` argument, `defaultTemplates`. This
```

Comment:
"new argument along side the `dimensions` argument, `defaultTemplates`" -> "new `defaultTemplates` argument alongside the `dimensions` argument".